src/schema/validator.js

```diff
@@ -409,9 +412,47 @@ export class SchemaValidator {
   }
 
   _compileAddonValidator(validator) {
-    const { minimum, maximum } = this.allowedManifestVersionsRange;
+    const { minimum, addon, maximum } = this.allowedManifestVersionsRange;
+
+    const replacer = (key, value) => {
+      if (Array.isArray(value)) {
+        value = value.filter((item) => {
+          let includeItem = true;
+          if (
+            item?.min_manifest_version &&
+            minimum < item.min_manifest_version
+          ) {
+            includeItem =
+              item.min_manifest_version >= minimum &&
+              item.min_manifest_version <= maximum &&
+              item.min_manifest_version <= addon;
+          }
+          if (
+            item?.max_manifest_version &&
+            maximum > item.max_manifest_version
+          ) {
+            includeItem =
+              item.max_manifest_version >= minimum &&
+              item.max_manifest_version <= maximum &&
+              item.max_manifest_version >= addon;
+          }
+
+          return includeItem;
+        });
+      }
+      return value;
+    };
+
+    // Omit from the schema data all entries that includes a
+    // min/max_manifest_version which is outside of the minimum
+    // and maximum manifest_version currently allowed per validator
+    // config and if they do not apply to the addon manifest_version.
+    //
+    const schemaObject = JSON.parse(
+      JSON.stringify(this.schemaObject, replacer, 2)
+    );
```

Comment:
@willdurand this is currently traversing the JSONSchema data by using JSON.stringify, which is likely not the best way to traverse the json objects given that it is stringifying all data and then parsing it again, I used it mainly as a quick way to give the approach a try and confirm if that would be a possible way to avoid these unexpected errors.

Besides the JSON.stringify trick, as I also mentioned in the PR description comment a better (and less expesive at runtime) approach may be to this kind of JSONSchema data filtering at import time (by creating a different set of JSONSchema data per manifest_version).

We may also consider looking into that approach now and see if we can get an alternative patch without a too big of an effort.

Let me know wdyt.

(in the PR description I tried to explain in a bit more detail what is going on in practice, but I'm happy to have a chat if that description didn't yet help you to get a good picture of the underlying issue).